paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/intent/IntentConfirmationInterceptor.kt

```diff
@@ -29,33 +27,44 @@ internal interface IntentConfirmationInterceptor {
     ): ConfirmationDefinition.Action<Args>
 
     interface Factory {
-        fun create(initializationMode: PaymentElementLoader.InitializationMode): IntentConfirmationInterceptor
+        suspend fun create(initializationMode: PaymentElementLoader.InitializationMode): IntentConfirmationInterceptor
     }
 
     companion object {
-        var createIntentCallback: CreateIntentCallback? = null
-
         const val COMPLETE_WITHOUT_CONFIRMING_INTENT = "COMPLETE_WITHOUT_CONFIRMING_INTENT"
-        const val PROVIDER_FETCH_TIMEOUT = 2
-        const val PROVIDER_FETCH_INTERVAL = 5L
     }
 }
 
 @OptIn(SharedPaymentTokenSessionPreview::class)
 internal class DefaultIntentConfirmationInterceptorFactory @Inject constructor(
+    private val deferredIntentCallbackRetriever: DeferredIntentCallbackRetriever,
     private val intentFirstConfirmationInterceptorFactory: IntentFirstConfirmationInterceptor.Factory,
     private val deferredIntentConfirmationInterceptorFactory: DeferredIntentConfirmationInterceptor.Factory,
     private val sharedPaymentTokenConfirmationInterceptorFactory: SharedPaymentTokenConfirmationInterceptor.Factory,
 ) : IntentConfirmationInterceptor.Factory {
-    override fun create(initializationMode: PaymentElementLoader.InitializationMode): IntentConfirmationInterceptor {
+    override suspend fun create(
```

Comment:
We're basically making the assumption that the user sets the right callback.

What if the user configures with shared payment tokens, but sets the confirmation token callback? Do we guard against this anywhere?

We're also assuming the user just does the right thing between the other 2 create intent callbacks. Which I think is fine, but the assumption might be worth calling out in a comment or something.